src/sourmash/commands.py

```diff
@@ -1228,21 +1229,42 @@ def multigather(args):
             result = None
 
             query_filename = query.filename
-            if not query_filename:
+            if not query_filename or query_filename == "-":
                 # use md5sum if query.filename not properly set
-                query_filename = query.md5sum()
+                output_base = query.md5sum()
+            elif args.output_add_query_md5sum:
+                # Uniquify the output file if all signatures were made from the same file (e.g. with --singleton)
+                assert query_filename and query_filename != "-"  # first branch
+                output_base = os.path.basename(query_filename) + "." + query.md5sum()
```

Comment:
what happens in the case of duplicated `md5sums`? I assume this will be much more likely with short sequences that are sketched with `--singleton`

Reply:
Then the `output_base` will be flagged as a duplicate - which I think is appropriate, if not ideal ;). People can choose to force allow override, or... do something else.